lib/statsd/instrument/assertions.rb

```diff
@@ -66,4 +67,8 @@ def assert_statsd_call(metric_type, metric_name, options = {}, &block)
       metric
     end
   end
+
+  def numeric?(object)
+    true if Float(object) rescue false
```

Comment:
This is both confusing (name and implementation do not match), and will probably break many tests where we use `sample_rate: 1` 

you can simply do `object.kind_of?(Numeric)`, and if you want a better check, add `(0.0..1.0).cover?(object)`